HW04.Rmd

```diff
@@ -69,51 +152,123 @@ Assume that:
 
 ```{r race}
-
+gun_death_s_h <- gun_deaths %>% 
+    select(race, intent) %>% 
+    filter(race == c("White", "Black", "Hispanic"), intent == c("Suicide", "Homicide")) %>% 
+    group_by(race) %>% 
+    count(intent) %>% 
+    mutate(total_death = sum(n)) %>% 
+    mutate(fraction_intent = n / total_death)
+
+gun_death_s_h %>% 
+    ggplot(aes(x = race, y = fraction_intent, fill = intent)) +
+    geom_col() +
+    labs(title = "proportion of Gun Deaths that are Homicides or Suicides by Race, 2012-2014",
+         x = "Race",
+         y = "Proportion") +
+    scale_y_continuous(expand = c(0,0))
 ```
 
-(hint maybe looking at percentages would be good)
-```{r race_2}
-
-```
+The graph shows that a White person killed by a gun is more likely to die by suicide, while a Black or Hispanic person who is killed by a gun is more likely to die by homicide.
 
 
 ### Are police-involved gun deaths significantly different from other gun deaths? Assess the relationship between police involvement and age, police involvement and race, and the intersection of all three variables.
 
 ```{r police}
-
-```
-
-Think about performing stats such as a t.test to see if differences you are plotting are statistically significant
-
-```{r stats?}
-
+police_factor <- c("Uninvolved", "Involved")
+gd_police_race <- gun_deaths %>% 
+    select(age, race, police) %>% 
+    filter(age != "Not Available") %>% 
+    group_by(race) %>% 
+    count(police) %>% 
+    mutate(total_death = sum(n)) %>% 
+    mutate(fraction = n / total_death)
+
+gd_police_race$police <- as.factor(rep.int(police_factor, 5)) #Change police column to factor
+
+gd_police_race2 <- gd_police_race %>% 
+    filter(police == "Involved")
+
+
+gd_police_race2 %>%  #Definitely not pretty, but it will do
+    ggplot(aes(x = reorder(race, fraction), y = fraction, fill = "red4")) +
+    geom_col() +
+    labs(title = "Fraction of Gun Death that Involve Police by Race",
+        y = "Fraction of Gun Deaths Involving Police",
+        x = "Race of Victim") +
+    theme(axis.text.x = element_text(angle = 20, hjust = 1)) + #There are packages that allow for the axis text to wrap, but I decided not to use it
+    theme(legend.position = "none")
 
 ```
 
 
 ```{r police_2}
+gd_police_age <- gun_deaths %>% 
+    select(age, police) %>% 
+    filter(age != "Not Available") %>% 
+    mutate(age = as.numeric(age), police = as.factor(police)) %>% 
+    group_by(age) %>% 
+    count(police) %>% 
+    mutate(total_death = sum(n)) %>% 
+    mutate(fraction = n / total_death)
+
+levels(gd_police_age$police) <- police_factor
+
+gd_police_age2 <- gd_police_age %>% 
+    filter(police == "Involved")
+
+gd_police_age2 %>% # missing columns represent zero police-involved deaths for that age
+    ggplot(aes(x = age, y = fraction, fill = "red4")) +
+    geom_col() +
+    labs(title = "Fraction of Gun Death that Involve Police by Age",

```

Comment:
If you ever want to display the fractions as percentages on your scales, just add the following line of code after ggplot(): scale_y_continuous(labels = scales::percent)